search_similar_entities/facetedSearchEndpoint.rb

```diff
@@ -0,0 +1,161 @@
+#QuerySimilar Endpoint
+
+#Builds complex query for"Query Similar" functionality; returns a link to the search page, with a preseeded query looking for all entities of type testEntity that share certain attributes with the current "testEntity" entity, or alternatively, the same query but with preselected facets
+
+# Verify params are OK
+raise "Missing required parameter: id" unless params[:id]
```

Comment:
Try to refrain from throwing exceptions for expected things.  Exceptions should be reserved for exceptional errors (things you truly did not expect).

Basically, it's better to return a human friendly error than a stack trace if you can.